flexmeasures/data/scripts/cli_tasks/db_pop.py

```diff
@@ -321,3 +322,51 @@ def create_power_forecasts(
         end_of_roll=pd.Timestamp(to_date).tz_localize(timezone)
         - timedelta(hours=horizon_hours),
     )
+
+
+@app.cli.command()
+def db_dump():
+    """Create a database dump of the database used by the app."""
+    db_uri = app.config.get("SQLALCHEMY_DATABASE_URI")
+    db_host_and_db_name = db_uri.split("@")[-1]
+    click.echo(f"Backing up {db_host_and_db_name} database")
+    command_for_dumping = f"pg_dump --no-privileges --no-owner --data-only --format=c --file=pgbackup_`date +%F-%H%M`.dump {db_uri}"
+    try:
+        proc = subprocess.Popen(command_for_dumping, shell=True)  # , env={
+        # 'PGPASSWORD': DB_PASSWORD
+        # })
+        proc.wait()
+        dump_success = 1
+
+    except Exception as e:
+        dump_success = 0
+        click.echo(f"Exception happened during dump: {e}")
+    if dump_success:
+        click.echo("db dump successful")
+    else:
+        click.echo("db dump unsuccessful")
+
+
+@app.cli.command()
+@click.argument("file", type=click.Path(exists=True))
+def db_restore(file: str):
+    """Restore the database used by the app, from a given database dump file."""
```

Comment:
So I ran `db-dump` and then `db-restore`, which won't really work, I guess because restoring into an existing db of the same name just produces duplicate key value violations. 

Or should `db-restore` drop the existing db before? In the dump file, I found `DROP DATABASE a1;` 

So I believe I'm confused what restoration means here. Maybe we can tell people in the docstring what to expect and what the best way to do here is (like making sure that there is no database of the same name yet?).

Reply:
In between you needed to run db-reset. I'll mention that in the docstring of db-restore. I'm just trying to automate some of the steps in [data/Readme.md](https://github.com/SeitaBV/flexmeasures/blob/main/flexmeasures/data/Readme.md) here.